test/server.js

```diff
@@ -23,14 +23,13 @@ describe('server', function() {
           rejectUnauthorized: false
         },
         res => {
-          if (res.statusCode !== 200) {
-            return reject(new Error('Request failed: ' + res.statusCode));
-          }
-
           res.setEncoding('utf8');
           let data = '';
           res.on('data', c => (data += c));
           res.on('end', () => {
+            if (Math.floor(res.statusCode / 100) !== 2) {
```

Comment:
Does this ever respond with a code not equal to 200?

Reply:
Yes. My new test case has 500 response

Reply:
if you ever respond with 201 or other 2xx you'll not need to update this line. I'm open to reverting back this to comparing strictly against 200 if you feel strongly about it